drivers/sensor/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2018 Roman Tataurov <diytronic@yandex.ru>
+ * Copyright (c) 2022 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Driver for DS18B20 1-Wire temperature sensors
+ * A datasheet is available at:
+ * https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf
+ */
+#define DT_DRV_COMPAT maxim_ds18b20
+
+#include <drivers/sensor.h>
+#include <drivers/sensor/w1_sensor.h>
+#include <kernel.h>
+#include <logging/log.h>
+#include <sys/__assert.h>
+
+#include "ds18b20.h"
+
+LOG_MODULE_REGISTER(DS18B20, CONFIG_SENSOR_LOG_LEVEL);
+
+static int ds18b20_configure(const struct device *dev);
+
+/* measure wait time for 9-bit, 10-bit, 11-bit, 12-bit resolution respectively */
+static const uint16_t measure_wait_ms[4] = { 94, 188, 376, 750 };
+
+static inline void ds18b20_temperature_from_raw(uint8_t *temp_raw,
+						struct sensor_value *val)
+{
+	uint16_t temp = sys_get_le16(temp_raw);
+
+	val->val1 = (temp & 0x0FFF) >> 4;
+	val->val2 = (temp & 0x000F) * 10000;
+}
+
+/*
+ * Write scratch pad, read back, then copy to eeprom
+ */
+static int ds18b20_write_scratchpad(const struct device *dev,
+				    struct ds18b20_scratchpad scratchpad)
+{
+	struct ds18b20_data *data = dev->data;
+	const struct device *bus = ds18b20_bus(dev);
+	uint8_t sp_data[4] = {
+		DS18B20_CMD_WRITE_SCRATCHPAD,
+		scratchpad.alarm_temp_high,
+		scratchpad.alarm_temp_low,
+		scratchpad.config
+	};
+
+	return w1_write_read(bus, &data->config, sp_data, sizeof(sp_data), NULL, 0);
+}
+
+static int ds18b20_read_scratchpad(const struct device *dev,
```

Comment:
@str4t0m isn't it `int`?

Reply:
I am not sure what you mean here. Do you want to say that I should check the return value of each ds18b20_read_scratchpad() call?

Reply:
I think the practice is to return unhandled error to the caller wherever possible, so I'm just suggesting that maybe you can return `ds18b20_read_scratchpad(dev, &data->scratchpad);` instead of 0

Reply:
Yes this make sense. I am now returning the status in line 140.